pinot-core/src/test/java/com/linkedin/pinot/core/segment/index/converter/SegmentV1V2ToV3FormatConverterTest.java

```diff
@@ -119,7 +122,17 @@ public void testConvert()
 
     FileTime afterLoadTime = Files.getLastModifiedTime(v3Location.toPath());
     Assert.assertEquals(afterConversionTime, afterLoadTime);
-  }
-
 
+    // verify that SegmentMetadataImpl loaded from segmentDirectory correctly sets
+    // metadata information after conversion. This has impacted us while loading
+    // segments by triggering download. That's costly. That's also difficult to test
+    Assert.assertFalse(new File(segmentDirectory, V1Constants.MetadataKeys.METADATA_FILE_NAME).exists());
+    SegmentMetadataImpl metaAfterConversion = new SegmentMetadataImpl(segmentDirectory);
+    Assert.assertNotNull(metaAfterConversion);
+    Assert.assertFalse(metaAfterConversion.getCrc().equalsIgnoreCase(String.valueOf(Long.MIN_VALUE)));
+    Assert.assertEquals(metaAfterConversion.getCrc(), beforeConversionMeta.getCrc());
```

Comment:
Is it possible to do positive checking here? Set a particular crc abd creation time and make sure that we get it after conversion